plugins/worker/server/__init__.py

```diff
@@ -86,8 +86,9 @@ def schedule(event):
 
         # Set the job status to queued and record the task ID from celery.
         job['celeryTaskId'] = asyncResult.task_id
-        ModelImporter.model('job', 'jobs').updateJob(
-            job, status=JobStatus.QUEUED)
+        jobModel = ModelImporter.model('job', 'jobs')
+        job = jobModel.save(job)
+        jobModel.updateJob(job, status=JobStatus.QUEUED)
```

Comment:
This strikes me as odd, we need to save the job before calling `updateJob`.  Is there no way to include the update to `job['celeryTaskId']` in `updateJob`?